master/internal/core_checkpoint.go

```diff
@@ -127,6 +152,50 @@ func (m *Master) getCheckpointImpl(
 	return nil
 }
 
+// @Summary Check for access to a checkpoint's files for download.
+// @Tags Checkpoints
+// @ID check-checkpoint
+// @Accept  json
+// @Produce  json
+// @Param   checkpoint_uuid path string  true  "Checkpoint UUID"
+// @Success 200 {} string ""
+//nolint:godot
+// @Router /checkpoints/{checkpoint_uuid}/access [get]
+func (m *Master) checkCheckpoint(c echo.Context) error {
+	args := struct {
+		CheckpointUUID string `path:"checkpoint_uuid"`
+	}{}
+	if err := api.BindArgs(&args, c); err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "invalid checkpoint_uuid: "+err.Error())
+	}
+	id, err := uuid.Parse(args.CheckpointUUID)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest,
+			fmt.Sprintf("unable to parse checkpoint UUID %s: %s",
+				args.CheckpointUUID, err))
+	}
+
+	curUser := c.(*detContext.DetContext).MustGetUser()
+	if err := m.canDoActionOnCheckpoint(c.Request().Context(), curUser, args.CheckpointUUID,
+		expauth.AuthZProvider.Get().CanGetExperimentArtifacts); err != nil {
+		s, ok := status.FromError(err)
+		if !ok {
+			return err
+		}
+		switch s.Code() {
+		case codes.NotFound:
+			return echo.NewHTTPError(http.StatusNotFound, s.Message())
+		case codes.PermissionDenied:
+			return echo.NewHTTPError(http.StatusForbidden, s.Message())
```

Comment:
Many secure systems would consider this an information leak, and would prefer to return a 404 to an unauthorized user.

I think since we are fetching by basically unguessable uuid's, that would only cause user confusion, and not offer any real security.  Nobody is going to try to check all possible UUIDs to see which ones are defined on a cluster, for instance.

But somebody from the RBAC team should probably offer guidance on this topic.

@ioga?

Reply:
unfortunately this is correct: we have to return 404 either way because security.